### PR TITLE
[ch-36869] Implement multistore selection UI

### PR DESCRIPTION
**Pull request by aldahick**
# Story Reference

[ch36869](https://app.clubhouse.io/ns8/story/36869)

## Summary

Turns out that doing this with Magento's built-in "UI Component" jazz would've been difficult-to-impossible anyways. Ended up using Select2 for sanity around the icon rendering. Questions and testing are welcome, most of this was implemented in the last couple of hours 😅 

## Author Checklist

I have added/updated:

- [ ] Tests
- [x] Documentation
- [x] Release notes (title of this PR)

## Version Bump

- [ ] Patch (bug fix - backwards compatible)
- [x] Minor (new functionality - backwards compatible)
- [ ] Major (significant change - not backwards compatible)

### CODE REVIEW

File: Block/Adminhtml/Header.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+class Header extends Template
+{
+    /** @var Config */
+    public $config;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context The context
```
Comment (davehensleyns8) [inline Block/Adminhtml/Header.php:30]: Please add the `$config` parameter.

File: Block/Adminhtml/Header.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public function __construct(
+        Config $config,
+        Context $context
+    ) {
+        parent::__construct($context);
+
+        $this->config = $config;
+    }
+
+    public function getStores(): array
```
Comment (davehensleyns8) [inline Block/Adminhtml/Header.php:41]: Please add a docblock to all public methods.

Comment (aldahick) [reply Block/Adminhtml/Header.php]: meant to delete that entire class 👍

Comment (davehensleyns8) [reply Block/Adminhtml/Header.php]: Guess that works too. 😸

File: Observer/Admin/DashboardInstantiation.php [hunk 3]
```diff
@@ -140,7 +140,13 @@ public function execute(Observer $observer) : void
                 'platformVersion' => (string) $this->productMetadata->getVersion()
             ];
 
-            $installResult = InstallerClient::install('magento', $installRequestData);
+            // phpcs:ignore
+            $devToken = getenv('NS8_DEV_TOKEN');
+            $installResult = [ 'accessToken' => $devToken ];
+            if (!$devToken) {
+                $installResult = InstallerClient::install('magento', $installRequestData);
+            }
```
Comment (rob-ns8) [inline Observer/Admin/DashboardInstantiation.php:148]: I like this a lot, but maybe the environment variable should be named `NS8_ACCESS_TOKEN` or something a bit less generic.

**Pull request merged**